shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -128,6 +129,13 @@ - (void)performCommonViewControllerInitialization {
   _statusBarStyle = UIStatusBarStyleDefault;
 
   [self setupNotificationCenterObservers];
+
+  flutter::Shell& shell = [_engine.get() shell];
+  shell.SetFirstFrameCallback([self] {
```

Comment:
I think we're creating a parallel path here. There is already a weak_platform_view->SetNextFrameCallback path in this file which I think does what we want.

Reply:
I looked at it a bit and it does seem like we could accomplish what we want with that.  I argue that it is more error prone since using it correctly depends on having knowledge about when a shell has a platform view, which isn't obvious from the FlutterViewController's perspective.

Reply:
Another problem is that FirstFrame semantics is not exactly the same as NextFrame.  Let's chat in person when you have a sec.

Reply:
If you have a FlutterViewController, the shell has a PlatformView, right?

Reply:
Maybe, there is at least a time where it doesn't since the FlutterViewController has to communicate the view to the shell.  It is harder to reason about than the other PR, that's why i prefer this instead: https://github.com/flutter/engine/pull/9952

Reply:
I prefer the other PR as well :D

Otherwise a setFlutterViewDidRenderCallback that only works conditionally is too obscure